spotipy/client.py

```diff
@@ -118,7 +118,7 @@ class Spotify(object):
     # instead of a more complex URL parsing function.
     #
     # [1] https://developer.spotify.com/documentation/web-api/#spotify-uris-and-ids
-    _regex_spotify_url = r'^(http[s]?:\/\/)?open.spotify.com\/(?P<type>track|artist|album|playlist|show|episode|user|audiobook)\/(?P<id>[0-9A-Za-z]+)(\?.*)?$'  # noqa: E501
+    _regex_spotify_url = r'^(http[s]?:\/\/)?open.spotify.com\/(intl-\w\w\/)?(?P<type>track|artist|album|playlist|show|episode|user|audiobook)\/(?P<id>[0-9A-Za-z]+)(\?.*)?$'  # noqa: E501
```

Comment:
If you can, let's add as a comment a regex101.com URL so we can quickly check which strings work with the regex. It would kind of serve as a test

Reply:
I tried to collect all possible types, but I couldn't find a link with `audiobook` in it. Not even the [Wayback Machine](https://web.archive.org/web/*/https://open.spotify.com/audiobook/*) gave a result.

I found this link here, but doesn't work either:
https://github.com/spotipy-dev/spotipy/blob/070b54f494474e17d3c3e91050d52b3b4a9b2c95/tests/integration/non_user_endpoints/test.py#L60

Reply:
probably fine for now as `intl` links seem to work!

Reply:
I'm not sure how to "implement" the comment. The other referenced links seem to be referenced by a description, but there's nothing much to describe about a regex101 link except that it can be used to validate the functionality.

Reply:
Hi, wouldnt a regex like `/intl[\D][a-z]*`  be more simple? Is the one I currently use for my projects where I use spotipy

Reply:
This is probably a somewhat objective statement, but I don't see why you think that your regex is simpler than mine. There are only two letters and therefore no fancy `\D` and similar additions are required. Just two `\w` and that's it.

Reply:
> This is probably a somewhat objective statement, but I don't see why you think that your regex is simpler than mine. There are only two letters and therefore no fancy `\D` and similar additions are required. Just two `\w` and that's it.

Yes but what if the internalization code changes in the future or even if theres a country that uses 3 letters?

Reply:
> Yes but what if the internalization code changes in the future or even if theres a country that uses 3 letters?
Spotify appears to use:
https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2
and not:
https://en.wikipedia.org/wiki/ISO_3166-1_alpha-3